File: java/src/main/java/mchehab/com/java/SensorsFragment.java

package mchehab.com.java;

import android.content.Context;
import android.hardware.Sensor;
import android.hardware.SensorManager;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.ArrayList;
import java.util.List;


public class SensorsFragment extends Fragment {

    public SensorsFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_sensors, container, false);
        ListView listView = view.findViewById(R.id.listView);
        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<>(getActivity(), android.R.layout
                .simple_list_item_1, getListSensors());
        listView.setAdapter(arrayAdapter);
        return view;
    }

    private List<String> getListSensors(){
        SensorManager sensorManager = (SensorManager) getActivity().getApplicationContext()
                .getSystemService(Context.SENSOR_SERVICE);
        List<Sensor> listSensors = sensorManager.getSensorList(Sensor.TYPE_ALL);
        List<String> listSensorsNames = new ArrayList<>();
        for(Sensor sensor:listSensors){
            listSensorsNames.add(sensor.getName());
        }
        return listSensorsNames;
    }
}

File: app/src/main/java/mchehab/com/readsensordata/SensorsFragment.kt
package mchehab.com.readsensordata

import android.content.Context
import android.hardware.Sensor
import android.hardware.SensorManager
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ListView

class SensorsFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_sensors, container, false)
        val arrayAdapter = ArrayAdapter<String>(activity!!.applicationContext, android.R.layout
                .simple_list_item_1, getListSensors())
        val listView = view.findViewById<ListView>(R.id.listView)
        listView.adapter = arrayAdapter
        return view
    }

    private fun getListSensors(): List<String>{
        val sensorManager = activity?.applicationContext?.getSystemService(Context.SENSOR_SERVICE)
                as SensorManager
        val sensorsList = sensorManager.getSensorList(Sensor.TYPE_ALL)
        val sensorsListNames = mutableListOf<String>()
        for(sensor in sensorsList){
            sensorsListNames.add(sensor.name)
        }
        return sensorsListNames
    }

}
